test/entities/swaps/v3/swap.integration.test.ts

```diff
@@ -0,0 +1,822 @@
+    const contextData = testDataObj[context];
+    if (!contextData || typeof contextData !== 'object') {
+        return undefined;
+    }
+
+    const contextDataObj = contextData as Record<string, unknown>;
+
+    // If subContext is provided, navigate to it
+    if (subContext) {
+        const subContextData = contextDataObj[subContext];
+        if (!subContextData || typeof subContextData !== 'object') {
+            return undefined;
+        }
+        const subContextDataObj = subContextData as Record<string, unknown>;
+        const swapKindKey =
+            swapKind === SwapKind.GivenIn ? 'GivenIn' : 'GivenOut';
+        return subContextDataObj[swapKindKey];
+    }
+
+    // No subContext, swapKind is directly under context
+    const swapKindKey = swapKind === SwapKind.GivenIn ? 'GivenIn' : 'GivenOut';
+    return contextDataObj[swapKindKey];
+}
+
+/**
+ * Sets test data in the nested structure.
+ * @param testData - The test data object to update
+ * @param testName - Name of the test case
+ * @param context - Test context (e.g., 'native input', 'permit2 direct approval')
+ * @param subContext - Optional sub-context (e.g., 'native output', 'token output')
+ * @param swapKind - The swap kind being tested
+ * @param data - The data to save
+ */
+function setTestData(
+    testData: Record<string, unknown>,
+    testName: string,
+    context: string,
+    subContext: string | undefined,
+    swapKind: SwapKind,
+    data: { queryOutput: unknown; call: unknown },
+): void {
+    if (!testData[testName]) {
+        testData[testName] = {};
+    }
+    const testDataObj = testData[testName] as Record<string, unknown>;
+
+    if (!testDataObj[context]) {
+        testDataObj[context] = {};
+    }
+    const contextDataObj = testDataObj[context] as Record<string, unknown>;
+
+    const swapKindKey = swapKind === SwapKind.GivenIn ? 'GivenIn' : 'GivenOut';
+
+    if (subContext) {
+        if (!contextDataObj[subContext]) {
+            contextDataObj[subContext] = {};
+        }
+        const subContextDataObj = contextDataObj[subContext] as Record<
+            string,
+            unknown
+        >;
+        subContextDataObj[swapKindKey] = data;
+    } else {
+        contextDataObj[swapKindKey] = data;
+    }
+}
+
+// ----------------------------------------------------------------------------
+// Setup Helpers
+// ----------------------------------------------------------------------------
+
+/**
+ * Sets up the fork and test client for integration tests.
+ * Configures token approvals for Permit2 and creates a snapshot for test isolation.
+ * @param test - The test configuration
+ * @returns Fork RPC URL, test client, and pre-approval snapshot
+ */
+async function setupForkAndClient(test: Test): Promise<{
+    fork: { rpcUrl: string };
+    client: PublicWalletClient & TestActions;
+    snapshotPreApprove: Hex;
+}> {
+    const fork = await startFork(test.anvilNetwork, jobId);
```

Comment:
I think this might be where the current issues come from - at least locally it seems like the jobId isn't unique for each test. So we have multiple tests running in parallel reseting/altering state on the same fork which causes issues with nonces, etc. I think the same might also be happening in a single test file where different tests run in parallel and reset to snapshot. Using `sequential` should stop that.